mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala

```diff
@@ -146,11 +151,29 @@ object Keyword {
       .toList match {
       // (class|trait|object) classname ext@@
       case (_: Token.Ident) :: (_: Token.Ident) :: kw :: Nil =>
-        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject]) true
+        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject] || kw.is[Token.KwEnum]) true
         else false
       // ... classname() ext@@
       case (_: Token.Ident) :: (_: Token.RightParen) :: _ => true
+      // ... classname[T] ext@@
+      case (_: Token.Ident) :: (_: Token.RightBracket) :: _ => true
+      case _ => false
+    }
+  }
+
+  private def derivesAfterExtends(leadingReverseTokens: Array[Token]): Boolean = {
+    leadingReverseTokens
+      .filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF])
+      .take(4)
+      .toList match {
+      // ... extends ... with A der@@
+      case (_: Token.Ident) :: (_: Token.Ident) :: (_: Token.KwWith) :: _ => true
+      // ... extends A(,|with) B der@@
+      case (_: Token.Ident) :: (_: Token.Ident) :: sep :: _ if sep.is[Token.KwWith] || sep.is[Token.Comma] => true
```

Comment:
```suggestion
      case (_: Token.Ident) :: (_: Token.Ident) :: sep :: (_: Token.KwExtends) if sep.is[Token.KwWith] || sep.is[Token.Comma] => true
```
?

Reply:
This would be `... extends with A der@@`, I don't think this case is usefull